packages/drawer/src/popup.ts

```diff
@@ -0,0 +1,228 @@
+            getStyle(document.body, 'paddingRight'),
+            10,
+          )
+        }
+        scrollBarWidth = getScrollBarWidth()
+        const bodyHasOverflow =
+          document.documentElement.clientHeight < document.body.scrollHeight
+        const bodyOverflowY = getStyle(document.body, 'overflowY')
+        if (
+          scrollBarWidth > 0 &&
+          (bodyHasOverflow || bodyOverflowY === 'scroll') &&
+          data.withoutHiddenClass
+        ) {
+          document.body.style.paddingRight =
+            data.computedBodyPaddingRight + scrollBarWidth + 'px'
+        }
+        addClass(document.body, 'el-popup-parent--hidden')
+      }
+    }
+
+    if (getComputedStyle(dom).position === 'static') {
+      dom.style.position = 'absolute'
+    }
+
+    dom.style.zIndex = String(PopupManager.nextZIndex())
+    data.opened = true
+
+    // XXX These code will be removed ？
+    // props.onOpen && onOpen();
+
+    doAfterOpen()
+  }
+
+  function doAfterOpen() {
+    _opening = false
+  }
+
+  function close() {
+    // XXX These code will be removed ？
+    // if (props.willClose && !props.willClose()) return;
+
+    // if (_openTimer !== null) {
+    //   clearTimeout(_openTimer);
+    //   _openTimer = null;
+    // }
+    // clearTimeout(_closeTimer);
+
+    // const closeDelay = Number(this.closeDelay);
+
+    // if (closeDelay > 0) {
+    //   _closeTimer = setTimeout(() => {
+    //     _closeTimer = null;
+    //     this.doClose();
+    //   }, closeDelay);
+    // } else {
+    doClose()
+    // }
+  }
+
+  function doClose() {
+    _closing = true
+
+    // XXX These code will be removed ？
+    // this.onClose && this.onClose();
+
+    if (props.lockScroll) {
+      setTimeout(restoreBodyStyle, 200)
+    }
+
+    data.opened = false
+
+    doAfterClose()
+  }
+
+  function doAfterClose() {
+    PopupManager.closeModal(_popupId)
+    _closing = false
+  }
+
+  function restoreBodyStyle() {
+    if (props.modal && data.withoutHiddenClass) {
+      document.body.style.paddingRight = data.bodyPaddingRight
+      removeClass(document.body, 'el-popup-parent--hidden')
+    }
+    data.withoutHiddenClass = true
+  }
+
+  onMounted(() => {
+    if (props.visible) {
+      data.rendered = true
+      open()
+    }
+  })
+
+  onBeforeMount(() => {
+    _popupId = 'popup-' + idSeed++
+    PopupManager.register(_popupId, {
+      ...vVm,
+      close,
+    } as ComponentPublicInstance<any>)
+  })
+
+  onBeforeUnmount(() => {
+    PopupManager.deregister(_popupId)
+    PopupManager.closeModal(_popupId)
+    restoreBodyStyle()
+  })
+
+  watch(
+    () => props.visible,
+    val => {
+      if (val) {
+        if (_opening) return
+        if (!data.rendered) {
+          data.rendered = true
+          nextTick(() => {
+            open()
+          })
+        } else {
+          open()
+        }
+      } else {
+        close()
+      }
+    },
+  )
+  return { popupData: data, open }
+}
+
```

Comment:
We need to replace this with `modalValue` A.K.A. `v-model`

Reply:
Like the above .Just simply replace it with 'modelValue‘?